compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -437,20 +451,22 @@ transform_dialect::ForeachThreadToWorkgroupOp::applyToOne(
     func::FuncOp target, SmallVectorImpl<Operation *> &results,
     transform::TransformState &state) {
   if (!isa<HAL::ExecutableOp, HAL::ExecutableVariantOp>(state.getTopLevel())) {
-    state.getTopLevel()->emitOpError(
+    // Definite failure.
+    return DiagnosedSilenceableFailure(state.getTopLevel()->emitOpError(
         "requires HAL::ExecutableOp or HAL::ExecutableVariantOp toplevel "
         "to attach the workgroup size information to a nested "
-        "ExecutableExportOp");
-    return DiagnosedSilenceableFailure(reportUnknownTransformError(target));
+        "ExecutableExportOp"));
   }
 
   IREE::HAL::ExecutableExportOp exportOp;
   state.getTopLevel()->walk([&](IREE::HAL::ExecutableExportOp op) {
     if (op.getSymName() == target.getName()) exportOp = op;
   });
   if (!exportOp) {
-    state.getTopLevel()->emitOpError("no IREE::HAL::ExecutableExportOp found");
-    return DiagnosedSilenceableFailure(reportUnknownTransformError(target));
+    results.assign(1, nullptr);
+    Diagnostic diag(target->getLoc(), DiagnosticSeverity::Remark);
```

Comment:
Nit: it doesn't have to be a remark (which may not be printed because _MLIR_ overrides that), warnings and errors are not emitted if the silenceable failure is silenced.